tests/flow/test_hnsw_tiered.py

```diff
@@ -0,0 +1,563 @@
+    num_elements = indices_ctx.num_labels
+    
+    threads_num = TieredIndex.get_threads_num()
+    
+    # Initialize time measurements to apply assert at the end.
+    tiered_index_time = 0
+    hnsw_index_time = 0
+    
+    # Create the tiered index
+    index = indices_ctx.tiered_index
+    
+    _, bf_dur, end_add_time = indices_ctx.populate_index(index)
+    
+    index.wait_for_index()
+    tiered_index_time = bf_dur + time.time() - end_add_time
+    
+    assert index.hnsw_label_count() == num_elements
+    
+    # Measure insertion to tiered index.
+    print(f"Insert {num_elements} insertion jobs took {round_ms(bf_dur)} ms")
+    print(f"Done insertion jobs executions using {threads_num} threads took {round_ms(tiered_index_time)} ms")
+    
+    # Measure total memory of the tiered index.
+    tiered_memory = bytes_to_mega(index.index_memory())
+    
+    print(f"total memory of tiered index = {tiered_memory} MB")
+    
+    hnsw_index = HNSWIndex(indices_ctx.hnsw_params)
+    _, hnsw_index_time, _ = indices_ctx.populate_index(hnsw_index)
+
+    print(f"Insert {num_elements} vectors to sync hnsw index took {round_(hnsw_index_time)} s")   
+    hnsw_memory = bytes_to_mega(hnsw_index.index_memory())
+    print(f"total memory of hnsw index = {hnsw_memory} MB")
+    
+    # The index memory should be at least as the total memory of the vectors.
+    assert hnsw_memory > indices_ctx.get_vectors_memory_size()
+    
+    # Tiered index memory should be greater than HNSW index memory.
+    assert tiered_memory > hnsw_memory
+    execution_time_ratio = hnsw_index_time / tiered_index_time
+    print(f"with {threads_num} threads, insertion runtime is {round_(execution_time_ratio)} times better \n")
+    
+
+def search_insert(is_multi: bool, num_per_label = 1):
+    indices_ctx = IndexCtx(data_size=100000, is_multi=is_multi, num_per_label=num_per_label)
+    index = indices_ctx.tiered_index
+
+    num_labels = indices_ctx.num_labels
+    
+    query_data = indices_ctx.generate_queries(num_queries=1)
+    
+    # Add vectors to the flat index.
+    bf_index = indices_ctx.init_and_populate_flat_index()
+    
+    # Start background insertion to the tiered index.
+    index_start, _, _ = indices_ctx.populate_index(index)
+    
+    correct = 0
+    k = 10
+    searches_number = 0
+    print(f"hnsw size = {index.hnsw_label_count()}")
+    # run knn query every 1 s. 
+    total_tiered_search_time = 0
+    prev_bf_size = num_labels
+    while index.hnsw_label_count() < num_labels:
+        # For each run get the current hnsw size and the query time.
+        bf_curr_size = index.get_curr_bf_size()
+        query_start = time.time()
+        tiered_labels, _ = index.knn_query(query_data, k)
+        query_dur = time.time() - query_start
+        total_tiered_search_time += query_dur
+        
+        print(f"query time = {round_ms(query_dur)} ms")
+        
+        # BF size should decrease.
+        print(f"bf size = {bf_curr_size}")
```

Comment:
```suggestion
        print(f"Running a query while indexing is done in the background took {round_ms(query_dur)} ms, while there are {bf_curr_size} vectors lest to index in the buffer")
        
        # BF size should decrease.
```

Reply:
This will be printed for each query, i think it's too long

Reply:
OK, you can make it shorter, but make it a bit more informative...

Reply:
I added a description before the while starts to run